node/subsystem-types/src/messages.rs

```diff
@@ -328,18 +328,13 @@ pub enum NetworkBridgeRxMessage {
 	NewGossipTopology {
 		/// The session info this gossip topology is concerned with.
 		session: SessionIndex,
-		/// Ids of our neighbors in the X dimensions of the new gossip topology,
-		/// along with their validator indices within the session.
-		///
-		/// We're not necessarily connected to all of them, but we should
-		/// try to be.
-		our_neighbors_x: HashMap<AuthorityDiscoveryId, ValidatorIndex>,
-		/// Ids of our neighbors in the X dimensions of the new gossip topology,
-		/// along with their validator indices within the session.
-		///
-		/// We're not necessarily connected to all of them, but we should
-		/// try to be.
-		our_neighbors_y: HashMap<AuthorityDiscoveryId, ValidatorIndex>,
+		/// Our validator index in the session, if any.
+		local_index: Option<ValidatorIndex>,
+		/// The canonical shuffling of validators for the session.
+		canonical_shuffling: Vec<(AuthorityDiscoveryId, ValidatorIndex)>,
```

Comment:
TBH it's not clear to me that the gossip-support subsystem needs to exist because it all can be derived from the session info. But this changeset gives us the flexbility we need for #5999

Reply:
I suppose it's true we can re-derive it. Looking up `SessionInfo` is kind of a nuisance.

Reply:
Further refactorings appreciated! But beyond the scope of this PR, IMO.

Reply:
I think it's reasonable to have it here to avoid another call for `SessionInfo`, this was mostly a question than a comment.